EIPS/bigint_modexp.md

```diff
@@ -0,0 +1,75 @@
+### Parameters
+
+* `GQUADDIVISOR: 20`
+
+### Specification
+
+At address 0x00......05, add a precompile that expects input in the following format:
+
+    <length_of_BASE> <length_of_EXPONENT> <length_of_MODULUS> <BASE> <EXPONENT> <MODULUS>
+    
+Where every length is a 32-byte left-padded integer representing the number of bytes to be taken up by the next value. Call data is assumed to be infinitely right-padded with zero bytes, and excess data is ignored. Consumes `floor(max(length_of_MODULUS, length_of_BASE) ** 2 * max(length_of_EXPONENT, 1) / GQUADDIVISOR)` gas, and if there is enough gas, returns an output `(BASE**EXPONENT) % MODULUS` as a byte array with the same length as the modulus.
```

Comment:
Would it be safe assuming that it is unlikely to have calls consuming gigabytes of data?

If yes, how about making each length field 32 bits wide or 96 bits in total:

```
     00000001
     00000020
     00000020
     03
     fffffffffffffffffffffffffffffffffffffffffffffffffffffffefffffc2e
     fffffffffffffffffffffffffffffffffffffffffffffffffffffffefffffc2f
```